### PR TITLE
Logger optimisation

### PR DESCRIPTION
**Pull request by DavideD**
@Sanne THese are the changes to the log to improve performance, I don't know if there is a JIRA for it so you will probably need to update the commit history.

I also left your original commit.

Cheers

### CODE REVIEW

File: engine/src/main/java/org/hibernate/search/query/engine/impl/AbstractHSQuery.java [hunk 1]
```diff
@@ -42,9 +43,9 @@
 import org.hibernate.search.spatial.DistanceSortField;
 import org.hibernate.search.spi.CustomTypeMetadata;
 import org.hibernate.search.spi.IndexedTypeIdentifier;
+import org.hibernate.search.spi.IndexedTypeMap;
```
Comment (Sanne) [inline engine/src/main/java/org/hibernate/search/query/engine/impl/AbstractHSQuery.java]: just curious: seems something special happened on this case?

**Comment by gunnarmorling** [comment]
Probably too late now, but you could have applied my automated refactoring for this: https://github.com/hibernate/hibernate-validator/pull/874. Maybe try and use it for OGM?

File: engine/src/main/java/org/hibernate/search/query/engine/impl/AbstractHSQuery.java (hunk 1 above)
Comment (DavideD) [reply engine/src/main/java/org/hibernate/search/query/engine/impl/AbstractHSQuery.java]: I probably organized the imports on this and didn't notice

**Comment by DavideD** [comment]
@gunnarmorling  I did the Hibernate OGM one and this one on Friday with regular expressions. Before I've seen your commit

**Comment by Sanne** [comment]
merged as HSEARCH-2935! thanks

**Pull request closed without merging**